notes_app/client/app/notes/notes_service.js

```diff
@@ -1,12 +1,40 @@
 (function() {
   'use strict';
 
-  notesService.$inject = ['$http'];
+  notesService.$inject = ['$http', '$log'];
 
-  function notesService($http) {
+  function notesService($http, $log) {
+    var api_path = 'http://localhost:8000/api/v1/note/';
     return {
-      'getList': function() {
-        return $http.get('http://localhost:8000/api/v1/note/');
+      'list': function() {
+        return new Promise(function(fulfill, reject) {
```

Comment:
Angular has it's own implementation of promises - https://docs.angularjs.org/api/ng/service/$q